client/my-sites/upgrades/domain-management/dns/dns-list.jsx

```diff
@@ -43,7 +43,11 @@ const DnsList = React.createClass( {
 	addDns: function( record ) {
 		addDnsAction( this.props.selectedDomainName, record, ( error ) => {
 			if ( error ) {
-				notices.error( error.message );
+				notices.error( error.message || this.translate( 'The DNS record could not be restored.' ) );
+			} else {
+				this.props.successNotice( this.translate( 'The DNS record has been restored.' ), {
+					duration: 5000
```

Comment:
@mtias Buggy image but case in point:
![image](https://cloud.githubusercontent.com/assets/991022/13083745/3670d686-d48b-11e5-82b4-be48c5d42182.png)

I understand where you're coming from and my argument isn't against what you're saying. But we're trying to solve one problem in another level (specifically: a design problem in framework level), which I'm not very fond of.